worlds/inscryption/Rules.py

```diff
@@ -0,0 +1,203 @@
+        return state.has("Caged Wolf Card", self.player)
+
+    def has_dagger(self, state: CollectionState) -> bool:
+        return state.has("Dagger", self.player)
+
+    def has_magnificus_eye(self, state: CollectionState) -> bool:
+        return state.has("Magnificus Eye", self.player)
+
+    def has_useful_act1_items(self, state: CollectionState) -> bool:
+        return state.has("Oil Painting's Clover Plant", self.player) and state.has("Squirrel Totem Head", self.player)
+
+    def has_all_epitaph_pieces(self, state: CollectionState) -> bool:
+        if self.world.options.epitaph_pieces_randomization.value == 0:
+            return state.has("Epitaph Piece", self.player, 9)
+        elif self.world.options.epitaph_pieces_randomization.value == 1:
+            return state.has("Epitaph Pieces", self.player, 3)
+        else:
+            return state.has("Epitaph Pieces", self.player, 1)
+
+    def has_camera_and_meat(self, state: CollectionState) -> bool:
+        return state.has("Camera Replica", self.player) and state.has("Pile Of Meat", self.player)
+
+    def has_monocle(self, state: CollectionState) -> bool:
+        return state.has("Monocle", self.player)
+
+    def has_obol(self, state: CollectionState) -> bool:
+        return state.has("Ancient Obol", self.player)
+
+    def has_epitaphs_and_forest_items(self, state: CollectionState) -> bool:
+        return self.has_camera_and_meat(state) and self.has_all_epitaph_pieces(state)
+
+    def has_act2_bridge_requirements(self, state: CollectionState) -> bool:
+        return self.has_camera_and_meat(state) or self.has_all_epitaph_pieces(state)
+
+    def has_tower_requirements(self, state: CollectionState) -> bool:
+        return self.has_act2_bridge_requirements(state) and self.has_monocle(state)
+
+    def has_inspectometer_battery(self, state: CollectionState) -> bool:
+        return state.has("Inspectometer Battery", self.player)
+
+    def has_gems_and_battery(self, state: CollectionState) -> bool:
+        return state.has("Gems Module", self.player) and self.has_inspectometer_battery(state)
+
+    def has_pelts(self, state: CollectionState, count: int) -> bool:
+        return state.has("Holo Pelt", self.player, count)
+
+    def has_one_pelt(self, state: CollectionState) -> bool:
+        return self.has_pelts(state, 1) and self.has_gems_and_battery(state)
+
+    def has_two_pelt(self, state: CollectionState) -> bool:
+        return self.has_pelts(state, 2) and self.has_gems_and_battery(state)
+
+    def has_three_pelt(self, state: CollectionState) -> bool:
+        return self.has_pelts(state, 3) and self.has_gems_and_battery(state)
+
+    def has_four_pelt(self, state: CollectionState) -> bool:
+        return self.has_pelts(state, 4) and self.has_gems_and_battery(state)
+
+    def has_five_pelt(self, state: CollectionState) -> bool:
+        return self.has_pelts(state, 5) and self.has_gems_and_battery(state)
```

Comment:
```suggestion
    def has_pelts(self, state: CollectionState, count: int) -> bool:
        return state.has("Holo Pelt", self.player, count) and self.has_gems_and_battery(state)
```
Having 5 separate has_num_pelt helpers feels excessive -- would recommend condensing them into one function.

Reply:
I agree that it's excessive. If I remember correctly, I think it was because my rules list is of the type dict[str, (CollectionState) -> bool] which doesn't allow me to add parameters like the number of pelts. If you know of a way, that would be great.

Reply:
Ah, well, I have no idea then -- this is probably fine then to keep using that kind of structure. I'm definitely no stranger to having to do some hacky thing to keep everything else smooth

Reply:
Have `has_pelts` return a lambda that captures the count

Then, instead, of `"Act 3 - Trader 1": self.has_one_pelt`` you would call the function to get a lambda to assign
`"Act 3 - Trader 1": self.has_pelts(1)`

Reply:
Pushed in the latest commit. Works wonderfully!